src/wallet/test/crypto_tests.cpp

```diff
@@ -80,8 +81,8 @@ bool OldDecrypt(const std::vector<unsigned char>& vchCiphertext, CKeyingMaterial
     if (fOk) fOk = EVP_DecryptUpdate(ctx, &vchPlaintext[0], &nPLen, &vchCiphertext[0], nLen) != 0;
     if (fOk) fOk = EVP_DecryptFinal_ex(ctx, (&vchPlaintext[0]) + nPLen, &nFLen) != 0;
     EVP_CIPHER_CTX_cleanup(ctx);
-
     EVP_CIPHER_CTX_free(ctx);
+    ERR_remove_thread_state(nullptr);
```

Comment:
Yes, the OpenSSL API is full of surprises! :-)

This is my understanding:

`OldDecrypt(…)` (in `crypto_tests.cpp`) [calls OpenSSL's `EVP_DecryptFinal_ex(…)`](https://github.com/practicalswift/bitcoin/blob/b3afb6242ec64b7c1f161ff6125b6c1faa801652/src/wallet/test/crypto_tests.cpp#L82).

If `EVP_DecryptFinal_ex(…)` encounters any errors [then `EVPerr(…)` is called](https://github.com/openssl/openssl/blob/64846096b18340b9a39ddd29a7a0e23c56f22959/crypto/evp/evp_enc.c#L515-L516).

[EVPerr(…) is defined](https://github.com/openssl/openssl/blob/354ab3653c0f9a218d91d4e5679c738896253d26/include/openssl/err.h#L103) as:

```
#define EVPerr(f,r)  ERR_PUT_error(ERR_LIB_EVP,(f),(r),OPENSSL_FILE,OPENSSL_LINE)
```

[`ERR_PUT_error(…)`](https://github.com/openssl/openssl/blob/354ab3653c0f9a218d91d4e5679c738896253d26/crypto/err/err.c#L368-L403) stores stores the errors in a thread local error queue for which the allocations are made using [`OPENSSL_zalloc(…)` in `ERR_get_state(…)`](https://github.com/openssl/openssl/blob/354ab3653c0f9a218d91d4e5679c738896253d26/crypto/err/err.c#L682-L707).

This change is not related to any cleanup in `~CInit()`. I'm just trying to hunt down some potential leaks in the tests to make Valgrind happy :-)